projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -508,990 +324,646 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
+    public decrement(datePart?: DatePart, delta?: number): void {
+        this.dateTimeEditor.decrement(datePart, delta);
     }
 
-    /**
-     * Clear the input field, date picker value and calendar selection.
-     *
-     * @hidden @internal
-     */
-    public clear(): void {
-        if (!this.disabled) {
-            this.isEmpty = true;
-            this.invalidDate = '';
-            this.deselectDate();
-            this._setCursorPosition(0);
+    //#region Control Value Accessor
+    /** @hidden @internal */
+    public writeValue(value: Date) {
+        this._value = value;
+        if (this.dateTimeEditor.value !== value) {
+            this.dateTimeEditor.value = this.value;
         }
     }
 
-    /**
-     * Evaluates when @calendar.onSelection event was fired
-     * and update the input value.
-     *
-     * @param event selected value from calendar.
-     *
-     * @hidden @internal
-     */
-    public handleSelection(date: Date): void {
-        if (this.value) {
-            date.setHours(this.value.getHours());
-            date.setMinutes(this.value.getMinutes());
-            date.setSeconds(this.value.getSeconds());
-            date.setMilliseconds(this.value.getMilliseconds());
-        }
-        const oldValue = this.value;
-        this.value = date;
+    /** @hidden @internal */
+    public registerOnChange(fn: any) {
+        this.onChangeCallback = fn;
+    }
+
+    /** @hidden @internal */
+    public registerOnTouched(fn: any) {
+        this.onTouchedCallback = fn;
+    }
 
-        this.emitValueChangeEvent(oldValue, this.value);
-        this.calendar.viewDate = date;
-        this.closeCalendar();
-        this.onSelection.emit(date);
+    /** @hidden @internal */
+    public setDisabledState?(isDisabled: boolean): void {
+        this.disabled = isDisabled;
     }
+    //#endregion
 
+    //#region Validator
     /** @hidden @internal */
-    public onOpenClick(event: MouseEvent) {
-        event.stopPropagation();
-        this.openDialog();
+    public registerOnValidatorChange(fn: any) {
+        this.onValidatorChange = fn;
     }
 
     /** @hidden @internal */
-    public onBlur(event, calcDate = true): void {
-        this._isInEditMode = false;
-        if (this.mode === InteractionMode.DropDown && calcDate) {
-            this.calculateDate(event.target.value, event.type);
+    public validate(control: AbstractControl): ValidationErrors | null {
+        const value = control.value;
+        const errors = {};
+        if (!value) {
+            Object.assign(errors, { value: true });
         }
+        if (value && this.disabledDates && isDateInRanges(value, this.disabledDates)) {
+            Object.assign(errors, { dateIsDisabled: true });
+        }
+        Object.assign(errors, DateTimeUtil.validateMinMax(value, this.minValue, this.maxValue));
 
-        if (this.collapsed) {
-            this._updateValidityOnBlur();
+        if (this.inputGroup && !this.inputGroup.isFocused) {
```

Comment:
Fairly certain you don't need to manipulate the input group state here, the whole point of validator integration is it will trigger the model state change we're hooked into already

Reply:
The validity is updated properly but the input group does not change its colors on its own if it has no focus.

Reply:
I see what you mean and I think this could be a little bug in the input group itself. Because right now if you focus the input group and then blur it and make it invalid, the proper style will be applied, however, if you have not focused it then, yes, the model will become invalid but no styles will be applied. I'll test it to be sure.

Reply:
Yeah, our editors are configured to show validation errors (through color) only after the model has been touched/is dirty and that's by design. If you modify the conditions with a value that's not entered by the user (not dirty) I think it's fairly reasonable to count this as app scenario that might as well update the value to a valid one or mark the model dirty if that;s the intention